kola/tests/ignition/filesystem.go

```diff
@@ -437,3 +446,42 @@ func testSwapActivation(c cluster.TestCluster) {
 		c.Fatalf("swap's size should be: %s, got %s", swapSize, size)
 	}
 }
+
+var rootDiskExtraPartition = conf.ContainerLinuxConfig(`
+storage:
+  disks:
+  - device: /dev/disk/by-id/virtio-primary-disk
+    partitions:
+    - label: VAR
+      number: 10
+      start: '9GiB'
+  filesystems:
+  - name: var
+    mount:
+      device: /dev/disk/by-partlabel/VAR
```

Comment:
For `/var` we should also make sure that the mount point is set up in the initrd with Ignition because systemd-tmpfiles should populate `/var` from the initrd. I think with CLC and the translation to Ignition v3 this is complicated and if you want to test `/var` we should rather use Butane. If it's not really about `/var` (or `/home` and similar) but an arbitrary mount point then it's easier to use CLC.

Reply:
> For /var we should also make sure that the mount point is set up in the initrd with Ignition because systemd-tmpfiles should populate /var from the initrd.

we don't populate `/var` with `systemd-tmpfiles` from the initrd, and ignition unmounts mounted partitions in one of the last stages. It does this to ensure a system does not depend on ignition performing the mount.

> I think with CLC and the translation to Ignition v3 this is complicated and if you want to test /var we should rather use Butane. If it's not really about /var (or /home and similar) but an arbitrary mount point then it's easier to use CLC.

It's not about the mountpoint at all, it's about adding partitions to a partitioned disk (that's why this was in `partitions.go`).

For sure we need a testcase that ensures we don't break backwards compat with ignition v2/CLC. `/var` is also a location users are likely to try to put on a separate filesystem.

Reply:
We do, that's in `base_image_var.conf` but yeah, seems that we have a problem in general with separate `/var` partitions, at least on the first boot before the final system also had a chance to set up the files in `/var`. What one might see as result is either a failed unit or a misbehaving system, e.g., the first boot journal might not be persisted (this was a bug we fixed by letting the initrd populate `/var` correctly).

Reply:
FYI: The topic of separate `/var` is also relevant for applying sysext images from the initrd. The suggestion from Lennart was to establish a kernel command line setting similar to what systemd has for `/usr`. It would then be properly mounted in `/sysroot/var`.

Reply:
What we can also do in Flatcar is to define a partition or filesystem label to expect and then add a mount unit for that (it can be compatible with a future systemd cmdline way by having this mount unit only be set up if the cmdline option is not there, I think we have similar logic for `/usr`).

Reply:
Started something for supporting separate `/var`: https://github.com/flatcar/bootengine/pull/75

Reply:
Would be good to write some files here with Ignition. We also have to add one more test case with Butane an defining the `/var` path in the initrd and writing some files because the uplifted v3 path is generated and can't be controlled whereas with Butane we can use the `/sysroot/var` mount point and have to check that this works.

Reply:
so i added the file creation part, but not the butane yet